matrix_sdk/src/client.rs

```diff
@@ -646,6 +649,93 @@ impl Client {
         self.send(request).await
     }
 
+    /// Search the homeserver's directory of public rooms.
+    ///
+    /// Returns a `get_public_rooms::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * `limit` - The number of `PublicRoomsChunk`s in each response.
+    ///
+    /// * `since` - Pagination token from a previous request.
+    ///
+    /// * `server` - The name of the server, if `None` the requested server is used.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// use matrix_sdk::Client;
+    /// # use url::Url;
+    /// # let homeserver = Url::parse("http://example.com").unwrap();
+    /// # let limit = Some(10);
+    /// # let since = Some("since token");
+    /// # let server = Some("server name");
+    ///
+    /// let mut cli = Client::new(homeserver).unwrap();
+    /// # use futures::executor::block_on;
+    /// # block_on(async {
+    /// assert!(cli.public_rooms(
+    ///     limit,
+    ///     since,
+    ///     server
+    /// ).await.is_ok());
+    /// # });
+    /// ```
+    pub async fn public_rooms(
+        &self,
+        limit: Option<u32>,
+        since: Option<&str>,
+        server: Option<&str>,
+    ) -> Result<get_public_rooms::Response> {
+        let limit = limit.map(|n| UInt::try_from(n).ok()).flatten();
+        let since = since.map(ToString::to_string);
+        let server = server.map(ToString::to_string);
+
+        let request = get_public_rooms::Request {
+            limit,
+            since,
+            server,
+        };
+        self.send(request).await
+    }
+
+    /// Search the homeserver's directory of public rooms with a filter.
+    ///
+    /// Returns a `get_public_rooms_filtered::Response`, an empty response.
+    ///
+    /// # Arguments
+    ///
+    /// * `room_search` - The easiest way to create this request is using the `RoomListFilterBuilder`.
+    ///
+    /// # Examples
+    /// ```
+    /// # use std::convert::TryFrom;
+    /// # use matrix_sdk::{Client, RoomListFilterBuilder};
+    /// # use matrix_sdk::api::r0::directory::get_public_rooms_filtered::{self, RoomNetwork, Filter};
+    /// # use url::Url;
+    /// # let homeserver = Url::parse("http://example.com").unwrap();
+    /// # let mut rt = tokio::runtime::Runtime::new().unwrap();
+    /// # rt.block_on(async {
+    /// # let last_sync_token = "".to_string();
+    /// let mut client = Client::new(homeserver).unwrap();
+    ///
+    /// let generic_search_term = Some("matrix-rust-sdk".to_string());
+    /// let mut builder = RoomListFilterBuilder::new();
+    /// builder
+    ///     .filter(Filter { generic_search_term, })
+    ///     .since(last_sync_token)
+    ///     .room_network(RoomNetwork::Matrix);
+    ///
+    /// client.public_rooms_filtered(builder).await.is_err();
```

Comment:
Should I remove the `asserts` from all the examples? The i`s_err` is a leftover from doc-tests that actually ran, and had to `assert` an error to avoid a large amount of setup in the doc-test.

Reply:
I think that makes sense, but if you're going to do that, please in a separate PR.